packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1049,64 +1010,157 @@ export class KubernetesClient {
         }
-        const groupVersion = this.groupAndVersion(manifest.apiVersion);
-        const namespaceToUse = manifest.metadata?.namespace || optionalNamespace || DEFAULT_NAMESPACE;
-        if (groupVersion.group === '') {
-          const client = ctx.makeApiClient(CoreV1Api);
-          await this.createV1Resource(client, manifest, namespaceToUse);
-        } else if (groupVersion.group === 'apps') {
-          const k8sAppsApi = this.kubeConfig.makeApiClient(AppsV1Api);
-          if (manifest.kind === 'Deployment') {
-            await k8sAppsApi.createNamespacedDeployment(namespaceToUse, manifest);
-          } else if (manifest.kind === 'DaemonSet') {
-            await k8sAppsApi.createNamespacedDaemonSet(namespaceToUse, manifest);
-          }
-        } else if (groupVersion.group === 'networking.k8s.io') {
-          // Add networking object support (Ingress for now)
-          const k8sNetworkingApi = this.kubeConfig.makeApiClient(NetworkingV1Api);
-          if (manifest.kind === 'Ingress') {
-            await k8sNetworkingApi.createNamespacedIngress(namespaceToUse, manifest);
+        try {
+          // try to get the resource, if it does not exist an error will be thrown and we will
+          // end up in the catch block
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          await client.read(spec as any);
+          // we got the resource, so it exists: patch it
+          //
+          // Note that this could fail if the spec refers to a custom resource. For custom resources
+          // you may need to specify a different patch merge strategy in the content-type header
+          //
+          // See: https://github.com/kubernetes/kubernetes/issues/97423
+          if (action === 'apply') {
+            const response = await client.patch(spec);
+            created.push(response.body);
           }
-        } else {
-          const client = ctx.makeApiClient(CustomObjectsApi);
-          const apiResource = await this.getAPIResource(client, groupVersion, manifest.kind);
-          await this.createCustomResource(
-            client,
-            groupVersion.group,
-            groupVersion.version,
-            apiResource.name,
-            apiResource.namespaced ? namespaceToUse : undefined,
-            manifest,
-          );
+        } catch (error) {
+          // we did not get the resource, so it does not exist: create it
+          const response = await client.create(spec);
+          created.push(response.body);
         }
       }
-      return undefined;
-    } catch (error) {
-      telemetryOptions = { error: error };
+      return created;
+    } catch (error: unknown) {
+      telemetryOptions.error = error;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if ((error as any)?.response?.body) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const err: any = error as { response: { body: unknown } };
```

Comment:
I think that if we avoid any line 1150 it's not to introduce it line 1152

we should probably do
```ts
if (error instanceOf HttpError) {
  const httpError = error as HttpError;
  if (httpError.response.body.message) {
  ...
```

Reply:
Fixed. Thanks, that is much better.